service/src/main/java/org/cbioportal/genome_nexus/service/internal/VerifiedGenomicLocationAnnotationServiceImpl.java

```diff
@@ -139,8 +139,8 @@ private VariantAnnotation verifyOrFailAnnotation(VariantAnnotation annotation)
         }
         LOG.debug("verifying providedReferenceAllele : '" + providedReferenceAllele + "'");
         String responseReferenceAllele = getReferenceAlleleFromAnnotation(annotation);
-        if (responseReferenceAllele.length() != providedReferenceAllele.length()) {
-            // for altered length Deletion-Insertion responses, recover full reference allele with followup query
```

Comment:
Any reason it's for `Deletion-Insertion` only?

Reply:
@sheridancbio thoughts?

Reply:
The main purpose of this verification is to make sure that if the user query is specifying reference allele nucleotides that they match the nucleotides in the reference genome. The real concern here is about deletions ... so if the user query says "deleted nucleotides ACC at positions 34782938-34782940" we wanted to not give annotation based on removing (from the reference genome which was different from what the user was basing their variant calls on) nucleotides TAG present at those positions in the VEP database reference genome. There were some cases encountered like this in annotations for AACR Genie samples I believe. Events represented as "insertion-deletions" also involve deletions so they were covered here as well.

Pure insertions basically specify no reference genome nucleotides (in our MAF format at least), so they are not of concern here. SNPs do specify a reference genome nucleotide ... so it would make sense to expand this logic (of checking the reference allele against the VEP database reference genome) I think. When we receive the response from VEP for SNPs I think it includes both the reference allele and the variant allele in the response so we would not need to make a follow up query .. we could instead compare the reference genome nucleotide to the nucleotide specified in the user query. I think adding that check would be good. It may be the case that when we convert MAF into genome region format for SNP variants we lose the reference allele details. I think I remember that some simplification in the formats involved dropping the reference allele ... so maybe that is why we cannot test it here (because the input to genome nexus has dropped the SNP reference nucleotide to conform to the updated format).